5.7.2

MadMax3947 opened the pull request:
* River of Skulls rework: Fewer enemies (except on EASY since there were very few enemies on that difficulty anyway), and increased focus on navigating the map.
* A couple balancing changes.
* Bugfixes and polishing changes.

MadMax3947 committed:
S17: improved mermen-naga rubber-banding, remove extraneous naga unit types

MadMax3947 committed:
S11: add discoverable dwarven map, rebalancing

MadMax3947 committed:
.pot update

MadMax3947 committed:
S11: change pre-map display of hidden coordinates

MadMax3947 committed:
S11: hide enemy sides from status table until sighted

MadMax3947 committed:
S11: centralize super shroud logic, storeroom deaths trigger death events

MadMax3947 committed:
S11: ensure that a road tile is reachable from anywhere in the spider web so drakes can't get stuck

MadMax3947 committed:
S11: balancing changes

MadMax3947 committed:
S11: strip metadata from map image

MadMax3947 committed:
Properly place super shroud after unit deaths

MadMax3947 committed:
Place super shroud after other die events

MadMax3947 committed:
S12: use lua gettext plural

MadMax3947 committed:
Pull out image dialog code from S11 into common.lua

MadMax3947 committed:
S17B: add icon for food menu item

MadMax3947 committed:
Part 2

MadMax3947 committed:
Add attack icon to fungoid stick, fix crash with fungoid unit description

MadMax3947 committed:
Add ability to override endlevel message, currently used for B branch losses that feed into S2a

MadMax3947 committed:
Fix endlevel message translatableness

MadMax3947 committed:
Appease wmllint again

MadMax3947 committed:
Fix bigmap journey icon locations

MadMax3947 committed:
Misc fixes

MadMax3947 committed:
S2B: make betrayal event less unfair

MadMax3947 committed:
Capitalize unit names in descriptions
Align with wesnoth PR 9892

MadMax3947 committed:
Avoid unnecessary warning from override_endlevel_message

MadMax3947 committed:
S15: remove gate sound after gate is destroyed

MadMax3947 committed:
S2: fix broken objectives screen after scenario end

MadMax3947 committed:
5.7.2

cooljeanius commented:
That's a lot of files to review...

cooljeanius commented on maps/drake11a.map:
```diff
@@ -43,7 +43,7 @@ Xu, Xu, Xu, Xu, Xu, Xu, Uu, Ur, Uu, Xu, Xu, Uu, Uu, Ur, Uu, Xu, Xu, Uu, Uu, Xu,
 Xu, Xu, Xu, Xu, Xu, Xu, Uu, Rd, Uu, Uu, Uu, Xu, Uu, Ur, Ur, Uu, Ur, Uu, Uu, Uu, Rd, Rd, Rd, Uu, Uu, Xu, Ur, Ur, Uu, Re, Xu, Xu, Xu, Uu, Uu, Uu, Ur, Ur, Uu, Uu, Xu, Xu
 Xu, Xu, Xu, Xu, Xu, Xu, Uu, Uu, Uu, Ur, Ur, Uu, Ur, Uu, Uu, Uu, Ur, Uu, Ur, Ur, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Xu, Uu, Uu, Xu, Uu, Rd, Uu, Ur, Uu, Uu, Xu, Xu, Xu, Xu, Xu
-Xu, Xu, Xu, Xu, Xu, Uu, Xu, Xu, Uu, Uu, Uu, Ur, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Xu, Uu, Uu, Rd, Uu, Ur, Uu, Re, Uu, Uu, Yl, Uu, Uu, Uu^Vu, Xu, Xu
+Xu, Xu, Xu, Xu, Xu, Uu, Xu, Xu, Uu, Uu, Uu, Ur, Uu, Uu, Uu, Ur, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Uu, Xu, Uu, Uu, Rd, Uu, Ur, Uu, Re, Uu, Uu, Yl, Uu, Uu, Uu^Vu, Xu, Xu
```
Extra road in this scenario is good; why just this one tile, though?

MadMax3947 replied:
This prevents non-quick burner-line drakes from getting stuck in the spider web. They require 6 mp to reach a cave tile (3 * 2 from slowing), so this one road tile serves as an off-ramp.

MadMax3947 committed:
Update for AWaS 1.3.0
Remove our own hacks to hide AWaS menu item in scenarios without weather, since newer AWaS version handles it

cooljeanius commented on images/storyline/s11_map.jpg:
So, will this image have to be updated if/when the map's actual terrain is updated? Is the procedure for doing that written down anywhere?

MadMax3947 replied:
This image is really just a placeholder. Ideally we'd ask someone who can draw on discord or forums to draw a  map that actually looks like a dwarf drew it by hand, and wouldn't necessarily be tile-level detailed.

MadMax3947 committed:
S8B: fix defeat conditions list

MadMax3947 committed:
Magi code overhaul

MadMax3947 committed:
Reduce redraw calls in magi target overlay code

cooljeanius commented on macros/bigmap.cfg:
current review progress: up to here, 199/231 files viewed

MadMax3947 committed:
Register magi attacks as synced commands

cooljeanius commented on scenarios/17a_Blockade.cfg:
```diff
@@ -438,15 +437,103 @@
     [/event]
 
-    #so the mermen last a bit longer
+    #rubber-banding to keep both sides competitive
```
hm I'm not sure how I feel about this; my usual strategy for this scenario is to wait for one of the sides to kill the other and then just mop up, and I'd be kind of frustrated if I couldn't do that anymore...

MadMax3947 replied:
The intended approach is for the player to use their units to screen mermen/nagas to permit the Unforgiver to slip through. However sitting back and waiting for one side to prevail, or for both sides to weaken each other enough for the player to stand a chance against them, are also certainly valid strategies.

To that end, the adjustments to hit probability only affect mermen and nagas fighting each other. Player's attacks and defenses are kept fair, and moreover don't affect the overall mermen vs. naga hit bias so focusing on the mermen or nagas will still weaken that side as expected. Also, in 34665110f0d7b2b6486101cede9349b4bff9b1f1 added a cap to the amount of mermen/naga bonus gold per turn so their armies can't balloon as much early on, which should also make alternate strategies more viable later in the scenario (albeit with more turn counter pressure).

cooljeanius replied:
ok

cooljeanius commented on units/animals/Fungoid.cfg:
```diff
@@ -10,6 +10,15 @@
     hitpoints=3
     movement_type=smallfoot
+#ifver WESNOTH_VERSION == 1.19.8
+    #unit description crashes Wesnoth if movement cost greater than movement, issue #9822
```
this comment should link to wesnoth/wesnoth#9822; there is no such issue here in the FtF repo

MadMax3947 replied:
Clarified this in the comment.

cooljeanius reviewed (approved):
OK, looks mostly fine to me... I assume you've tested?

MadMax3947 committed:
Clarify issue reference

MadMax3947 committed:
S17a: cap mermen and naga gold bonus per turn

MadMax3947 committed:
S17a: cleanup store/recall code

MadMax3947 committed:
S2b: make traitor event only fire for level 1 unit

MadMax3947 committed:
Centralize continue after defeat [message]

MadMax3947 committed:
.pot update

MadMax3947 committed:
Use boolean_equals= instead of equals= properly

MadMax3947 committed:
Clarify variable name in S19 and S20

MadMax3947 committed:
S19: typo fix
